### PR TITLE
Document the TZDB update policy

### LINKED ISSUE #1042: Determine and document support lifetime policy
In particular, we *may* need the following to be handled differently:

- Minor (in impact) bugfixes
- Major (in impact) bugfixes
- TZDB updates

Initial proposal:

We expect users within a major version to update within a year of a new minor version becoming available. After that time, the previous minor version is unsupported. For example, 1.4.0 was released on July 8th 2017, so from July 8th 2018 we won't supply new patch versions of 1.3.x.

We can probably say that minor bugfixes will only be applied to the current minor version in each major version, whereas major bugfixes and TZDB updates will be applied to all the supported minor versions.

The point at which we decide not to support 1.x at all is a whole other decision.

### PR DESCRIPTION
**Pull request by jskeet**
Fixes #1042

### CODE REVIEW

File: src/NodaTime.Web/Markdown/root/versions.md [hunk 1]
```diff
@@ -708,3 +711,16 @@ Bug fixes:
 ## 1.0.0-beta1, released 2012-04-12 with tzdb 2012c
 
 - Initial beta release
+
+# <a name="tzdb-updates"></a>Update policy for time zone data
```
Comment (malcolmr) [inline src/NodaTime.Web/Markdown/root/versions.md]: prefer `<a id="">` to `<a name="">`.

(Also, I assume this actually DTRT in the rendered version?)

File: src/NodaTime.Web/Markdown/root/versions.md [hunk 2]
```diff
@@ -708,3 +711,16 @@ Bug fixes:
...
+
+New time zone data is released by IANA as required. The latest data
+is always available via the Noda Time web site as described in the
+[user guide](/userguide/tzdb). Additionally, NuGet packages are
```
Comment (malcolmr) [inline src/NodaTime.Web/Markdown/root/versions.md]: Maybe:

```
in the [user guide]().

Additionally, released versions of the NodaTime NuGet package are also updated...

(next para)
```

File: src/NodaTime.Web/Markdown/root/versions.md [hunk 3]
```diff
@@ -708,3 +711,16 @@ Bug fixes:
...
+updated with a patch release to include the latest data.
+
+The latest minor release for each major release is always updated.
```
Comment (malcolmr) [inline src/NodaTime.Web/Markdown/root/versions.md]: That's just a complicated way of saying "the latest release", isn't it? :)

File: src/NodaTime.Web/Markdown/root/versions.md [hunk 4]
```diff
@@ -708,3 +711,16 @@ Bug fixes:
...
+Additionally, if a minor release is less than 6 months old, the
```
Comment (malcolmr) [inline src/NodaTime.Web/Markdown/root/versions.md]: ", if the latest minor release is..."?

File: src/NodaTime.Web/Markdown/root/versions.md (hunk 1 above)
Comment (jskeet) [reply src/NodaTime.Web/Markdown/root/versions.md]: Didn't know about this. Will check both forms.

File: src/NodaTime.Web/Markdown/root/versions.md (hunk 2 above)
Comment (jskeet) [reply src/NodaTime.Web/Markdown/root/versions.md]: I think I'd rather not have two single-line paragraphs.

File: src/NodaTime.Web/Markdown/root/versions.md (hunk 3 above)
Comment (jskeet) [reply src/NodaTime.Web/Markdown/root/versions.md]: Nope - "1.4" isn't the latest release, but it's the latest minor release of a major release.

File: src/NodaTime.Web/Markdown/root/versions.md (hunk 4 above)
Comment (jskeet) [reply src/NodaTime.Web/Markdown/root/versions.md]: Yup

File: src/NodaTime.Web/Markdown/root/versions.md (hunk 3 above)
Comment (malcolmr) [reply src/NodaTime.Web/Markdown/root/versions.md]: I'm apparently unable to read. Thanks for the clarification :)

**Pull request merged**